Source/DafnyRuntime/DafnyRuntime.py

```diff
@@ -17,13 +17,24 @@ def __get__(self, instance, owner):
 def print(value):
     builtins.print(string_of(value), end="")
 
+# Dafny strings are currently sequences of UTF-16 code units.
+# To make a best effort attempt at printing the right characters we attempt to decode,
+# but have to allow for invalid sequences.
+def string_from_utf_16(utf_16_code_units):
+    return b''.join([ord(c).to_bytes(2, 'little') for c in utf_16_code_units]).decode("utf-16", errors = 'replace')
+
 def string_of(value) -> str:
     if hasattr(value, '__dafnystr__'):
         return value.__dafnystr__()
     elif value is None:
         return "null"
     elif isinstance(value, bool):
         return "true" if value else "false"
+    elif isinstance(value, str):
+        # This is only for Dafny char values.
+        # Dafny strings are represented as Seq's of indivdual char values,
+        # and Seq defines __dafnystr__.
+        return string_from_utf_16(value)
```

Comment:
Do you need the full generality of this if it can only be the value of a Dafny char?

Reply:
I still need most of it at least, since I need to convert a single character to a `bytes` of length 2 to decode from UTF-16. Given that a Python "character" is just a string with length 1, there's no real overhead to using the same function as for strings.

Reply:
+1, thanks